packages/gatsby/src/internal-plugins/component-layout-creator/validate-path.js

```diff
@@ -0,0 +1,11 @@
+const systemPath = require(`path`)
+
+module.exports = path => {
+  // Disallow paths starting with an underscore
+  // and template-.
+  const parsedPath = systemPath.parse(path)
+  return (
+    parsedPath.name.slice(0, 1) !== `_` &&
```

Comment:
It's the convention that we don't treat underscored files in `src/pages` as pages so that's copied here. The idea being you could have helper components in the same directory if you wanted.

Perhaps not that useful of convention but...